src/pages/dataTableUtilities/getColumns.js

```diff
@@ -220,6 +251,15 @@ const PAGE_COLUMNS = {
     COLUMN_NAMES.COMMENT,
     COLUMN_NAMES.REMOVE,
   ],
+  [ROUTES.SUPPLIER_INVOICES_WITH_PROGRAMS]: [
+    COLUMN_NAMES.INVOICE_NUMBER,
+    COLUMN_NAMES.SUPPLIER,
+    COLUMN_NAMES.PROGRAM,
+    COLUMN_NAMES.ENTRY_DATE,
+    COLUMN_NAMES.TRANSACTION_TYPE,
+    COLUMN_NAMES.COMMENT,
+    COLUMN_NAMES.REMOVE,
+  ],
```

Comment:
> I think only SR and CR's should have the program column? 

I wasn't sure about the invoices in general, so I think you're probably right. I ended up doing them just to be safe and consistent with the issue: `Add a Programme column in the following list displays: Requisition, Invoices, Stocktakes`.

>  The only time a linked requisition is set (I could be wrong!) is from an incoming CR 

I think all incoming transactions have a linked requisition if they have a `requisition_ID`. Is that only CRs?

> - so I suppose it would show a program name for finalised CI's which were a program internal order from another store? 

Yeah, as I understand it, it goes as follows... please correct any of this which is wrong!

1. store makes SR to mobile
2. mobile makes response CR.
3. user finalises CR, creating finalised CI.

In this situation, the CI has the same program as the response CR, which is the program of the original SR. 

> And never anything for an SI...

Again, not an expert on this so correct where I'm wrong!

1. mobile makes SR to store.
2. store makes CR, CI and finalises.
3. mobile gets SI.

In this situation, the SI has the same program as the original SR.

> ...(so just a bunch of ugly n/a's) ?

That's ok, as if none of the records have a program, the column doesn't appear :smile:.

Reply:
> I wasn't sure about the invoices in general, so I think you're probably right. I ended up doing them just to be safe and consistent with the issue: Add a Programme column in the following list displays: Requisition, Invoices, Stocktakes.

I think if we're both happy to remove it from invoices, I reckon we should!

> That's ok, as if none of the records have a program, the column doesn't appear

I was actually thinking about this: There are potentially 1000's of records to check when navigating which isn't ideal. I think it's possibly better to just use `isProgram` on master lists to determine to show?

Reply:
Nice.

Re. performance of checking for programs, I haven't benchmarked but I'd take a guess that anything linear in the number of records isn't going to be too much of an issue. Just from my perspective, I think its more user friendly for the list to be a function of its contents, rather than the state of the database (e.g. if I have no program requisitions, rendering a column based on a masterlist I haven't used would be a bit counter-intuitive/confusing/frustrating) ? I'm not particularly worried either way but I think whatever is most user friendly is probably the best thing to focus on as the performance is likely to be a minor overhead, and if not, can probably be addressed by using a realm query in the page initialisation or something like that?

Reply:
I think it will definitely be a slight annoyance potentially adding a second maybe close to two for the navigation. Iterating over 5000+ elements in an array generally isn't a concern but in JS on a tablet it's not great